In [187]:
import pandas as pd


df = pd.read_csv("/drive2/kaggle/pii-dd/piidd/training/assertion/mistral-7b/3/oq88dwrd/20240227050913_preds.csv", engine="python")

In [188]:
print(len(df))
df

75


B-EMAIL  B-ID_NUM  B-NAME_STUDENT  B-PHONE_NUM  B-STREET_ADDRESS  \
0   2.5427878e-06  0.000268        0.971675     0.000005      2.749409e-06   
1   2.0538969e-06  0.000016        0.994090     0.000001      9.043184e-07   
2   3.0597832e-05  0.000449        0.157906     0.000014      6.438756e-06   
3    8.754587e-06  0.000010        0.000080     0.000001      2.204876e-06   
4   3.2823777e-06  0.000057        0.071317     0.000006      1.019969e-05   
..            ...       ...             ...          ...               ...   
70   0.0001089085  0.000858        0.476535     0.000063      2.534294e-05   
71  2.9631913e-06  0.000019        0.964752     0.000001      6.547520e-07   
72  8.4202824e-05  0.000549        0.000741     0.000038      1.692406e-05   
73   4.259247e-05  0.001014        0.000351     0.000012      2.051301e-06   
74   0.0009006546  0.000494        0.108593     0.000009      5.576461e-05   

    B-URL_PERSONAL  B-USERNAME       I-EMAIL      I-ID_NUM  I-NAME_STUDENT  \
0         0.000022    0.000589  2.670029e-06  5.719079e-06        0.001103   
1         0.000003    0.000004  1.994299e-07  6.707212e-07        0.000127   
2         0.000596    0.000194  6.413654e-06  4.803604e-06        0.000330   
3         0.000052    0.000066  3.495963e-06  1.746165e-05        0.703976   
4         0.000053    0.000048  1.093035e-06  3.597953e-06        0.000010   
..             ...         ...           ...           ...             ...   
70        0.000403    0.000992  9.866457e-06  4.725447e-05        0.001566   
71        0.000009    0.000039  6.308999e-07  4.634520e-06        0.000020   
72        0.281387    0.000456  1.275002e-05  5.264135e-05        0.000658   
73        0.068360    0.000768  5.565135e-06  4.334779e-05        0.000154   
74        0.000425    0.005594  5.147321e-05  3.131635e-04        0.247602   

    I-PHONE_NUM  I-STREET_ADDRESS  I-URL_PERSONAL    I-USERNAME         O  \
0      0.000011      6.417587e-06    4.749843e-07  4.419363e-06  0.026302   
1      0.000002      5.873031e-07    2.396206e-07  2.642017e-07  0.005752   
2      0.000011      4.227396e-05    7.182951e-06  3.379758e-06  0.840399   
3      0.000003      1.067410e-05    2.196280e-06  4.722745e-06  0.295762   
4      0.000005      5.649330e-06    2.763627e-07  1.165805e-06  0.928478   
..          ...               ...             ...           ...       ...   
70     0.000047      1.993079e-05    1.825384e-05  6.546854e-06  0.519299   
71     0.000006      1.386110e-06    6.598873e-07  1.109429e-06  0.035141   
72     0.000024      1.541936e-04    7.749556e-05  3.010703e-06  0.715746   
73     0.000063      7.667443e-05    5.827389e-05  4.325675e-06  0.929045   
74     0.000145      9.226886e-04    1.047943e-04  4.111825e-05  0.634748   

             label                                            context document  
0                O  solutions\t\r  to\t\r  mitigate\t\r  the\t\r  ...     9098  
1                O  Example Reflection – Storytelling (Design Tool...     7676  
2   B-NAME_STUDENT  mind? This ad was only the 3rd  most valuable ...     3885  
3                O  : In\n\nPerson, Online & On the Phone\n\nBuild...     1798  
4   B-NAME_STUDENT  \n\n1. Challenge\n\nDuring my Master Thesis in...     8747  
..             ...                                                ...      ...  
70  B-NAME_STUDENT   potential for gaining new clients and reprodu...      308  
71               O  Instructor: [START] Isabel [END] Yomna\n\nBY: ...     4600  
72  B-URL_PERSONAL  has actually collected real interviews with yo...     9297  
73  B-URL_PERSONAL  your own words, briefly describe the tool you ...    12456  
74  I-NAME_STUDENT  Classification : Giuseppe Petrosillo - Confide...     8340  

[75 rows x 18 columns]

In [192]:
pred_cols = list(df.columns)[:-3]
pred_cols

['B-EMAIL',
 'B-ID_NUM',
 'B-NAME_STUDENT',
 'B-PHONE_NUM',
 'B-STREET_ADDRESS',
 'B-URL_PERSONAL',
 'B-USERNAME',
 'I-EMAIL',
 'I-ID_NUM',
 'I-NAME_STUDENT',
 'I-PHONE_NUM',
 'I-STREET_ADDRESS',
 'I-URL_PERSONAL',
 'I-USERNAME',
 'O']

In [203]:
i = -1

In [214]:
# r = df.sample(n=1)
i += 1
r = df.iloc[i]

print(r.label)
print(r.document)

pairs = [(round(float(getattr(r, col)), 3), col) for col in pred_cols if float(getattr(r, col)) > 0.001]
for p in pairs:
    print(p)
    
print("*"*30)

print(r.context.replace("\r", "\n"))

B-NAME_STUDENT
4094
(1.0, 'O')
******************************
”

Written by Rosa Perna

MDI-191

07.17.15

How climate change affects the economy?

“The time is past when humankind

thought it could selfishly draw on

exhaustible resources. We know now

the world is not a commodity.”

[START] Leo [END] Ine

Climate change has always happened on Earth but since the Industrial

Revolution, human activity has been affecting the atmosphere, which has led to an  increase in temperature and caused global warming. These days such problem as  global warming is considered to 


In [201]:
[r.context]

["solutions\t\r \xa0to\t\r \xa0mitigate\t\r \xa0the\t\r \xa0tourism\t\r \xa0 challenge\t\r \xa0on\t\r \xa0the\t\r \xa0coastal\t\r \xa0region\t\r \xa0of\t\r \xa0the\t\r \xa0city\t\r \xa0of\t\r \xa0Rimini.\t\r \xa0Application,\t\r \xa0insights\t\r \xa0and\t\r \xa0approach\t\r \xa0are\t\r \xa0“fiction”.\t\r \xa0 \t\r \xa0 Cheers,\t\r \xa0 [START] Omar [END] \t\r \xa0Vulpinari\t\r \xa0 \t\r \xa0 \t\r \xa0 —CHALLENGE—\t\r \xa0 \t\r \xa0 I'm\t\r \xa0from\t\r \xa0Rimini,\t\r \xa0the\t\r \xa0most\t\r \xa0popular\t\r \xa0Italian\t\r \xa0seaside\t\r \xa0town\t\r \xa0spread\t\r \xa0along\t\r \xa034\t\r \xa0kilometres\t\r \xa0of\t\r \xa0Adriatic\t\r \xa0Sea\t\r \xa0 coastline.\t\r \xa0From\t\r \xa0the\t\r \xa060s"]

doc 


gubqlvufez
22088 (Reyes)
12255 (Millette)
huokbdyvcb (username)
jfsobckrsw (username)
2722 (Hector)
13144 (gdelossantos)
13249 (David)
21527 (I-NAME_STUDENT)
21580 (Christopher)
16541 (Jose)
9098 (Omar)
22109 (Hamzah)

In [14]:
import json
import os
from dotenv import load_dotenv
from pathlib import Path

dotenv_path = Path("../../.env")
if dotenv_path.exists():
    print("Loaded .env file!")
    load_dotenv(str(dotenv_path))


data = json.load(open(Path(os.environ["PROJECT_HOME_DIR"]) / "data/train.json"))

In [47]:
d = [d for d in data if d["document"] == 11387][0]
d["full_text"]

'Challenge: Describe your challenge, including all relevant information.\n\nThe September/October 2004 issue of CG&A introduced the term visual analytics to the  computer science literature.1 In 2005, an international team of multidisciplinary panelists  consensually and collectively defined the then newly established area as “the science of  analytical reasoning facilitated by interactive visual interfaces.”2 The means and targets of  VA have since evolved and expanded significantly, covering both scientific and nonscientific  data of different types, shapes, sizes, domains, and applications. As extreme-scale datasets  began revolutionizing our daily working life,3 researchers looked to VA for solutions to their  big-data problems.\n\nToday’s extreme-scale VA applications often combine high-performance computers for  computation, high-performance database appliances and/or cloud servers for data storage and  management, and desktop computers for human-computer interaction. Sources of 

In [24]:
print(d["tokens"])

['Elements', ':', '\n\n', '1', '.', 'Challenge', ':', 'There', 'are', 'many', 'challenges', 'which', 'one', 'can', 'face', 'as', 'a', 'teacher', '.', 'The', 'basic', 'job', 'of', 'a', '\n\n', 'teacher', 'is', 'to', 'teach', 'the', 'young', 'minds', 'and', 'inspire', 'them', ',', 'and', 'this', 'job', 'is', 'best', 'done', 'when', 'the', 'first', ' ', 'student', 'from', 'your', 'class', 'to', 'the', 'last', 'student', 'is', 'benefited', 'from', 'what', 'you', 'teach', '.', 'Even', 'if', 'one', ' ', 'student', 'is', 'left', 'out', 'the', 'task', 'a', 'teacher', 'is', 'assigned', 'to', 'do', 'is', 'not', 'accomplished', '.', 'This', 'challenge', ' ', 'came', 'to', 'me', 'when', 'I', 'had', '2', 'students', 'in', 'my', 'class', 'who', 'were', 'differently', 'able', '.', 'These', '2', 'students', ' ', 'were', 'a', 'part', 'of', 'inclusive', 'education', 'program', 'of', 'my', 'college', 'and', 'they', 'both', 'were', 'sponsored', 'by', 'a', ' ', 'big', 'financial', 'company', ',', 'as', 'a'